---
layout: page
title: Genome analysis tools, software and code

tools:
- name        : SequenceServer - implement, visualize & interpret BLAST analyses
  img         : sequenceserver.png
  url         : http://sequenceserver.com
  about       : "BLAST is the most commonly used bioinformatics tool. But setting it up for private data and using it is counter-intuitive. We developed SequenceServer to make BLAST easy to use. Used in hundreds of laboratories worldwide."
  publication_url : https://doi.org/10.1093/molbev/msz185
  publication_txt : "Sequenceserver: a modern graphical user interface for custom BLAST databases. Mol Biol Evol (2019)."

- name        : GeneValidator - choose the best gene models
  img         : genevalidator.png
  url         : https://wurmlab.github.io/tools/genevalidator
  code        : https://github.com/wurmlab/genevalidator
  about       : Tool to identify problematic gene predictions based on comparison with gene predictions in public databases. Funded by <a href="http://informatics.nescent.org/wiki/Phyloinformatics_Summer_of_Code_2013">NESCent Google Summer of Code 2013</a>. Useful for manual annotation, and for automatically choosing between multiple gene predictions for a given locus.
  publication_url : http://doi.org/10.1093/bioinformatics/btw015
  publication_txt : "GeneValidator: identify problems with protein-coding gene predictions. Bioinformatics (2016)."

- name        : TGNet - evalutate genome quality with an independent transcriptome
  img         : tgnet.jpg
  code        : https://github.com/ksanao/TGNet
  about       : <a href="https://github.com/ksanao/TGNet">TGNet</a> is a <a href="http://www.cytoscape.org">Cytoscape</a>-based tool for visualization and quality assessment of de novo genome assemblies. Specifically it facilitates rapid detection of inconsistencies between a genome assembly and an independently derived transcriptome assembly. Developed by <a href="http://scholar.google.co.uk/citations?user=aovYPpkAAAAJ&hl=en">Oksana Riba-Grognuz</a>. Used for <a href="http://www.pnas.org/content/early/2011/01/24/1009690108">fire ant genome</a> project.
  publication_url : http://bioinformatics.oxfordjournals.org/content/27/24/3425
  publication_txt : "Visualization and quality assessment of <em>de novo</em> genome assemblies. Bioinformatics (2011)."

- name        : Fourmidable / antgenomes.org
  img         : antgenomes.png
  url         : http://antgenomes.org
  about       : Community database centralizing genomic and transcriptomic sequence information for ants and related species (e.g., bees and wasps). Provides BLAST search and data download. Updated periodically with latest data and new tools.
  publication_url : /publications/wurm2009antDatabase.pdf
  publication_txt : "Fourmidable: a database for ant genomics. BMC Genomics 10:5 (2009)."

- name        : BioNode
  img         : tools/bionode.png
  url         : http://bionode.io
  code        : http://bionode.io
  about       : <a href="http://bionode.io">BioNode</a> is <a href="http://nodejs.org">Node.js</a> JavaScript library for <a href="http://bionode.io">client and server side bioinformatics</a>.

oldtools: 
- name        : oswitch
  img         : oswitch.png
  url         : https://github.com/wurmlab/oswitch
  code        : https://github.com/wurmlab/oswitch
  about       : Provides quick and simple Docker based virtual environments on Macs and compute clusters alike for reproducible bioinformatics. Funded by <a href="http://environmentalomics.org/portfolio/big-data-infrastructure/">NERC Environmental 'omics (EOS) Cloud</a>. Given improvements to Docker this isn't really needed anymore. 

- name        : Afra
  img         : afra.png
  code        : http://github.com/wurmlab/afra
  about       : <a href="http://bbsrc.ac.uk">BBSRC</a>-funded tool for crowd-sourcing curation of gene predictions.
  status      : Beta. Used in Bioinforatics MSc course at QMUL.
  status_icon : fa-warning

---
<h1>Major bioinformatic software tools that we develop include:</h1>
{% for tool in page.tools %}
<div class="row">
        <div class="col-sm-2 col-md-2 hidden-xs">
            <img src="{{ site.url }}/img/{{ tool.img }}" alt="{{ tool.name }}" style="height: 125px; margin-top: 15px;" class="img-rounded center-block img-responsive">
        </div>
        <div class="col-sm-10 col-md-9">
            <div class="caption">
                <h3><a href="{{ tool.url }}">{{ tool.name }}</a></h3>
                <ul class="fa-ul" style="text-align: justify;">
                    <!-- tool status and about is compulsory -->
                    <li>
                        <i class="fa-li fa fa-info-circle"></i>{{ tool.about }}
                    </li>
                    {% if tool.publication_url %}
                    <li>
                        <i class="fa-li fa fa-file-text"></i><a href="{{ tool.publication_url}}">{{ tool.publication_txt }}</a>
                    </li>
                    {% endif %} {% if tool.code %}
                    <li>
                        <i class="fa-li fa fa-code"></i><a href="{{tool.code}}">{{ tool.code}}</a>
                    </li>
                    {% endif %}
                    {% if tool.status %}
                    <li>
                        <i class="fa-li fa {{tool.status_icon}}"></i>{{ tool.status }}
                    </li>
                    {% endif %}
                </ul>
            </div>
        </div>
        <div class="col-sm-2 col-md-2 visible-xs-inline ">
            <img src="{{ site.url }}/img/{{ tool.img }}" alt="{{ tool.name }}" style="height: 125px; margin-top: 15px;" class="img-rounded center-block img-responsive">
        </div>
        <div class="clearfix"></div>
</div>
{% endfor %}

<br/><br/>

<h2>Software development approaches</h2>
<p>We believe in <a href="http://www.nature.com/nature/focus/reproducibility/">Reproducible Research</a> &amp; <a href="http://software.ac.uk">Software Sustainability</a>. When possible we respect style guides and use unit testing &amp; continuous integration. We aim to respect these approaches in our research on specific biological questions and in our tool development. We develop mostly in:</p>
<ul>
    <li>
        <a href="http://rproject.org">R</a> for number crunching.
    </li>
    <li>
        Shell for minimalisting pipelining.
    </li>
    <li>
      Python or <a href="http://bioruby.org">ruby</a> when text needs to be analysed. 
    </li>
    <li>
        javascript/html/css for client-side stuff and some <a href="http://bionode.io">bioinformatics analysis pipelines</a>
    </li>
    <li>
        But also use bash/python/perl when convenient or necessary...
    </li>
</ul>
<p>But we love quick and dirty hacks too <i class="fa fa-smile-o"></i>.</p>
<p>A lot of our code is at <a href="http://github.com/wurmlab">http://github.com/wurmlab</a>.</p>

<br/><br/>

<h2>Some tools now have limited relevance</h2>
<p>oSwitch has been superceded by updates in docker. Afra has been used in our MSc course; it is currently somewhat out of sync with main Apollo. </p>
{% for tool in page.oldtools %}
<div class="row">
        <div class="col-sm-2 col-md-2 hidden-xs">
            <img src="{{ site.url }}/img/{{ tool.img }}" alt="{{ tool.name }}" style="height: 150px; margin-top: 15px;" class="img-rounded center-block img-responsive">
        </div>
        <div class="col-sm-10 col-md-9">
            <div class="caption">
                <h3><a href="{{ tool.url }}">{{ tool.name }}</a></h3>
                <ul class="fa-ul" style="text-align: justify;">
                    <!-- tool status and about is compulsory -->
                    <li>
                        <i class="fa-li fa fa-info-circle"></i>{{ tool.about }}
                    </li>
                    {% if tool.publication_url %}
                    <li>
                        <i class="fa-li fa fa-file-text"></i><a href="{{ tool.publication_url}}">{{ tool.publication_txt }}</a>
                    </li>
                    {% endif %} {% if tool.code %}
                    <li>
                        <i class="fa-li fa fa-code"></i><a href="{{tool.code}}">{{ tool.code}}</a>
                    </li>
                    {% endif %}
                    {% if tool.status %}
                    <li>
                        <i class="fa-li fa {{tool.status_icon}}"></i>{{ tool.status }}
                    </li>
                    {% endif %}
                </ul>
            </div>
        </div>
        <div class="col-sm-2 col-md-2 visible-xs-inline ">
            <img src="{{ site.url }}/img/{{ tool.img }}" alt="{{ tool.name }}" style="height: 150px; margin-top: 15px;" class="img-rounded center-block img-responsive">
        </div>
        <div class="clearfix"></div>
</div>
{% endfor %}


